app/bundles/CoreBundle/Config/config.php

```diff
@@ -1177,7 +1177,7 @@
         'db_user'                         => '',
         'db_password'                     => '',
         'db_table_prefix'                 => '',
-        'db_server_version'               => '5.7',
+        'db_server_version'               => defined('MAUTIC_DB_NOT_CONFIGURED') ? '5.7' : null,
```

Comment:
This is a crucial part of the logic. When no `config/local.php` has been created yet, `MAUTIC_DB_NOT_CONFIGURED` is defined. See PR description for more details.

Reply:
@alanhartless I might need some help here. As correctly indicated by @RCheesley from her testing, the `db_server_version` parameter is still written to `config/local.php` this way. Earlier on in my journey for this PR, I removed `db_server_version` from `CoreBundle/Config/config.php` and added the logic to `app/config/config.php`, here:

https://github.com/mautic/mautic/blob/c6b7ae73c6b0c701e21aba5fa068b840a45e0017/app/config/config.php#L117

However that seems to write the value into Symfony's cache, which we also don't want. Because otherwise it remains at `5.7` even after installation, until folks remove their cache. Is there a way to dynamically get the value from `db_server_version` here (as it does now) while preventing it from being written to `config/local.php`? That's the final missing piece in this PR then!

Reply:
Idea: we might be able to leverage `MAUTIC_CONFIG_PARAMETERS` for this (I couldn't find any documentation about that env variable by the way, but looking at the code we might be able to set/update it to `{"db_server_version": "5.7"}` in the AppKernel, instead of setting that constant). That way, we can remove `db_server_version` from `CoreBundle/Config/config.php` and it won't be stored in `config/local.php` anymore 🚀 

I'll test this theory later today

https://github.com/mautic/mautic/blob/7dbb6ee8dc5753ee1ac94f7e8bdae6cd650cb2e0/app/bundles/CoreBundle/Loader/ParameterLoader.php#L160-L166

Reply:
FIXED IN https://github.com/mautic/mautic/pull/9739/commits/9f8bc8a4961610040343ae7b9b0182477fab5d36 🎉 🚀 CC @RCheesley @alanhartless